code/game/objects/items/weapons/RCD.dm

```diff
@@ -382,3 +389,285 @@ RCD
 	name = "admin RCD"
 	max_matter = INFINITY
 	matter = INFINITY
+
+
+/obj/item/weapon/rcd/arcd
+	name = "advanced rapid-construction-device (ARCD)"
+	max_matter = 300
+	matter = 300
+	color = "red"
+
+
+// Ranged RCD
+
+
+/obj/item/weapon/rcd/arcd/afterattack(atom/A, mob/user, proximity)
+	proximity = 1
+	if(!(A in view(7, get_turf(user))))
+		to_chat(user, "<span class='warning'>The \'Out of Range\' light on the RCD blinks red.</span>")
+		return 0
+	if(isturf(A) || istype(A, /obj/machinery/door/airlock) || istype(A, /obj/structure/grille) || istype(A, /obj/structure/window) || istype(A, /obj/structure/girder))
```

Comment:
This is to control the beam, not the actual RCD. My extensive local testing confirmed that being able to spam beams on invalid targets was a bad idea.